src/OrchardCore/OrchardCore.DisplayManagement/Implementation/DefaultHtmlDisplay.cs

```diff
@@ -74,13 +75,13 @@ public async Task<IHtmlContent> ExecuteAsync(DisplayContext context)
                 var shapeTable = _shapeTableManager.GetShapeTable(theme?.Id);
 
                 // Use the same prefix as the shape
-                //var originalHtmlFieldPrefix = context.ViewContext.ViewData.TemplateInfo.HtmlFieldPrefix;
-                //context.ViewContext.ViewData.TemplateInfo.HtmlFieldPrefix = shapeMetadata.Prefix ?? "";
+                var viewContext = context.ServiceProvider.GetService<ViewContextAccessor>().ViewContext;
```

Comment:
Meh, can you explain how to reproduce the issue with the previous code?
And no way to use the DisplayContext to forward the prefix until it's really necessary to use a ViewContext?

Reply:
Repro with the previous code:

- Use the blog setup, then edit an article and publish.

- The body is lost because the html body part was not prefixed.

Anyway we were creating a `localContext` which was never used.

What i saw is that when we were creating a new `ViewContext` based on an existing one, the new one was still sharing the same `ViewData` and then `TemplateInfo.HtmlFieldPrefix`. So, anyway at the end we need to restore the original prefix in `viewContext.ViewData.TemplateInfo.HtmlFieldPrefix`.

Reply:
@sebastienros 

I will try to redo exactly as we did but when creating a new viewContext i will try to also pass a new viewData based on the previous one. Looking at the code, this way it seems that we keep the same model state but we don't share the same data.